imap_processing/codice/codice_l1a.py

```diff
@@ -566,55 +566,125 @@ def set_data_product_config(
+    by epoch, then by priority, then by events. For example, for a CoDICE-Lo
+    dataset with 10 epochs, we expect the length of the `event_data` field to
+    be (10 epochs * 8 priorities) = 80 items, with each item being a compressed
+    byte object representing a variable number of events (up to 10000 events).
+    Each compressed byte object is comprised of several fields with specific
+    bit lengths/positions, described by the constants.[LO|HI]_DE_BIT_STRUCTURE
+    dictionary. Padding is added to any fields that have less than 10000 events.
+
+    In order to process these data, we must take the decommed raw data, group
+    the packets appropriately based on their `seq_flgs`, decompress the data,
+    then arrange the data into CDF data variables for each priority and bit
+    field. For example, P2_SpinAngle represents the spin angles for the 2nd
+    priority data.
 
     Parameters
     ----------
     apid : int
         The APID of the packet.
-    packet : xarray.Dataset
-        The packet to process.
+    packets : xarray.Dataset
+        The packets to process.
     data_version : str
         Version of the data product being created.
 
     Returns
     -------
     dataset : xarray.Dataset
-        Xarray dataset containing the event data.
+        Xarray dataset containing the direct event data.
     """
+    # Set some useful variables unique to CoDICE-Lo and CoDICE-Hi
     if apid == CODICEAPID.COD_LO_PHA:
-        dataset_name = "imap_codice_l1a_lo-pha"
+        num_priorities = 8
+        cdf_fields = [
+            "NumEvents",
+            "DataQuality",
+            "APDGain",
+            "APD_ID",
+            "APDEnergy",
+            "TOF",
+            "MultiFlag",
+            "PHAType",
+            "SpinAngle",
+            "EnergyStep",
+        ]
     elif apid == CODICEAPID.COD_HI_PHA:
-        dataset_name = "imap_codice_l1a_hi-pha"
+        num_priorities = 6
+        cdf_fields = [
+            "NumEvents",
+            "DataQuality",
+            "SSDEnergy0,TOF",
+            "SSD_ID",
+            "ERGE",
+            "MultiFlag",
+            "Type",
+            "SpinAngle",
+            "SpinNumber",
+        ]
 
-    # Extract the data
-    # event_data = packet.event_data.data (Currently turned off, see TODO)
+    # Group and decompress the data
+    grouped_data = group_data(packets)
+    decompressed_data = [
+        decompress(group, CoDICECompression.LOSSLESS) for group in grouped_data
+    ]
+
+    # Reshape the packet data into CDF-ready variables
+    data = reshape_de_data(packets, decompressed_data, num_priorities)
 
+    # Gather the CDF attributes
     cdf_attrs = ImapCdfAttributes()
     cdf_attrs.add_instrument_global_attrs("codice")
     cdf_attrs.add_instrument_variable_attrs("codice", "l1a")
     cdf_attrs.add_global_attribute("Data_version", data_version)
 
     # Define coordinates
     epoch = xr.DataArray(
-        packet.epoch,
+        packets.epoch[::num_priorities],
```

Comment:
Can you add a comment about why you're doing this? Should num_priorities be another dimension of the dataset?